services/dynamic-sidecar/tests/unit/test_api_containers_tasks.py

```diff
@@ -0,0 +1,604 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+
+import json
+from collections import namedtuple
+from contextlib import asynccontextmanager, contextmanager
+from inspect import getmembers, isfunction
+from typing import (
+    Any,
+    AsyncIterable,
+    AsyncIterator,
+    Awaitable,
+    Callable,
+    Final,
+    Iterator,
+    Optional,
+)
+
+import aiodocker
+import faker
+import pytest
+from _pytest.fixtures import FixtureRequest
+from aiodocker.containers import DockerContainer
+from aiodocker.volumes import DockerVolume
+from asgi_lifespan import LifespanManager
+from fastapi import FastAPI
+from fastapi.routing import APIRoute
+from httpx import AsyncClient
+from pydantic import AnyHttpUrl, parse_obj_as
+from pytest_mock.plugin import MockerFixture
+from servicelib.fastapi.long_running_tasks.client import (
+    Client,
+    TaskClientResultError,
+    TaskId,
+    periodic_task_result,
+)
+from servicelib.fastapi.long_running_tasks.client import setup as client_setup
+from simcore_sdk.node_ports_common.exceptions import NodeNotFound
+from simcore_service_dynamic_sidecar._meta import API_VTAG
+from simcore_service_dynamic_sidecar.api import containers_tasks
+from simcore_service_dynamic_sidecar.models.shared_store import SharedStore
+
+FAST_STATUS_POLL: Final[float] = 0.1
+CREATE_SERVICE_CONTAINERS_TIMEOUT: Final[float] = 60
+DEFAULT_COMMAND_TIMEOUT: Final[int] = 5
+
+ContainerTimes = namedtuple("ContainerTimes", "created, started_at, finished_at")
+
+
+# UTILS
+
+
+def __print_routes(app: FastAPI) -> None:
+    endpoints = []
+    for route in app.routes:
+        if isinstance(route, APIRoute):
+            endpoints.append(route.path)
+
+    print("ROUTES\n", json.dumps(endpoints, indent=2))
+
+
+def _get_dynamic_sidecar_network_name() -> str:
+    return "entrypoint_container_network"
+
+
+@contextmanager
+def mock_tasks() -> Iterator[None]:
+    # NOTE: mocking via `mocker: MockerFixture` appears not to work
```

Comment:
hmm shall we have a look together at this one?

Reply:
Had another look, nailed this one down.